ptf/sainat.py

```diff
@@ -0,0 +1,1493 @@
+        # Translated server IP - 20.20.20.1
+        # WAN IP - 10.10.10.1
+        # self.server_ip = '192.168.0.1'
+
+        l4_src_port = 100
+        l4_dst_port = 1234
+        proto = 6
+
+        translate_sport = 500
+        translate_dport = 2000
+
+        nat_type = SAI_NAT_TYPE_DESTINATION_NAT
+        nat_data = sai_thrift_nat_entry_data_t(
+            key=sai_thrift_nat_entry_key_t(
+                dst_ip=self.translate_server_ip,
+                proto=proto,
+                l4_dst_port=l4_dst_port))
+        dnat1 = sai_thrift_nat_entry_t(vr_id=self.default_vrf,
+                                       data=nat_data,
+                                       nat_type=nat_type)
+        sai_thrift_create_nat_entry(self.client,
+                                    nat_entry=dnat1,
+                                    nat_type=nat_type,
+                                    dst_ip=self.server_ip,
+                                    l4_dst_port=translate_dport)
+        self.nat_list.append(dnat1)
+
+        nat_data = sai_thrift_nat_entry_data_t(
+            key=sai_thrift_nat_entry_key_t(
+                dst_ip=self.translate_server_ip))
+        dnat2 = sai_thrift_nat_entry_t(vr_id=self.default_vrf,
+                                       data=nat_data,
+                                       nat_type=nat_type)
+        sai_thrift_create_nat_entry(self.client,
+                                    nat_entry=dnat2,
+                                    nat_type=nat_type,
+                                    dst_ip=self.server_ip)
+        self.nat_list.append(dnat2)
+
+        nat_type = SAI_NAT_TYPE_DESTINATION_NAT_POOL
+        nat_data = sai_thrift_nat_entry_data_t(
+            key=sai_thrift_nat_entry_key_t(
+                dst_ip=self.translate_server_ip))
+        dnat3 = sai_thrift_nat_entry_t(vr_id=self.default_vrf,
+                                       data=nat_data,
+                                       nat_type=nat_type)
+        sai_thrift_create_nat_entry(self.client,
+                                    nat_entry=dnat3,
+                                    nat_type=nat_type,
+                                    dst_ip=self.server_ip)
+        self.nat_list.append(dnat3)
+
+        nat_type = SAI_NAT_TYPE_SOURCE_NAT
+        nat_data = sai_thrift_nat_entry_data_t(
+            key=sai_thrift_nat_entry_key_t(
+                src_ip=self.server_ip,
+                proto=proto,
+                l4_src_port=l4_src_port))
+        snat4 = sai_thrift_nat_entry_t(vr_id=self.default_vrf,
+                                       data=nat_data,
+                                       nat_type=nat_type)
+        sai_thrift_create_nat_entry(self.client,
+                                    nat_entry=snat4,
+                                    nat_type=nat_type,
+                                    src_ip=self.translate_server_ip,
+                                    l4_src_port=translate_sport)
```

Comment:
Do you want to add a comment here to reflect this is same as sai attribute: SAI_NAT_ENTRY_ATTR_SRC_IP and SAI_NAT_ENTRY_ATTR_L4_SRC_PORT